hlx_statics/blocks/onthispage/onthispage.js

```diff
@@ -45,25 +61,35 @@ export default async function decorate(block) {
 
         if (maxVisibleSection) {
             const id = maxVisibleSection.id;
-            const correspondingAnchor = Array.from(anchors).find(anchor => anchor.getAttribute('href') === `#${id}`);
+            const correspondingAnchor = anchors.find(anchor => anchor.getAttribute('href') === `#${id}`);
             if (correspondingAnchor) {
                 anchors.forEach(anchor => anchor.classList.remove('active'));
                 correspondingAnchor.classList.add('active');
             }
         }
     }, {
         threshold: 0.3,
-        rootMargin: '-20% 0px -20% 0px'
+        rootMargin: '-20% 0px -20% 0px',
     });
 
-    sections.forEach(section => observer.observe(section));
+    headings.forEach(section => observer.observe(section));
 
     anchors.forEach(anchor => {
         anchor.addEventListener('click', (event) => {
+            isClick = true;
+            const anchorId = event.target.href.split('#')[1];
             anchors.forEach(a => a.classList.remove('active'));
             event.target.classList.add('active');
+            document.getElementById(anchorId).scrollIntoView({ behavior: 'smooth' });
+            setTimeout(() => {
+                isClick = false;
+            }, 500);
         });
     });
-    applyAnalyticHeaderOverride(block);
```

Comment:
Do we wanna remove analytics this for OnThisPage?

Reply:
@timkim do you know if we need this call to `applyAnalyticHeaderOverride`? I remember Dmitry mentioning before that this is how we know analytics has been applied to a block. I also see other blocks calling it: https://github.com/search?q=repo%3AAdobeDocs%2Fadp-devsite%20applyAnalyticHeaderOverride&type=code

Reply:
Ah yes we do need it.